cmd/main.go

```diff
@@ -113,7 +113,12 @@ func main() {
 				glog.Infof("launching watcher for ConfigMaps")
 				err := configWatcher.Watch(ctx, sigChan)
 				if err != nil {
-					glog.Fatalf("error watching for new ConfigMaps (terminating): %s", err.Error())
+					switch err {
+					case watcher.EventNilObjectError:
+						glog.Errorf("watcher got error, try to restart watcher: %s", err.Error())
```

Comment:
don't forget to `watcher.Stop()` here, so resources get cleaned up before the next iteration of the `for` loop calls `Watch()` again to restart.